### ISSUE TITLE
Toolkit Feature: Add back short/broad histology mapping

### ISSUE THREAD

**Issue by aadamk** [d3b-center/d3b-analysis-toolkit#60]
# ✨

#### [Required] Category.

- [ ] New tool
- [ ] Bug Fix
- [ ] Update to existing tool
- [ ] Documentation Update

#### [Required] Name of tool.

short/broad histology mapping

#### [Required] Upcoming deadlines or milestones.

2/10

#### [Required] Please detail the required features, updates, work to be done, etc.

Add back short/broad histology mapping, as it is no longer within `main`. The `add_short_and_broad_histologies.py` script is missing from `main` and is no longer in `annotate_histologies.py`: 

https://github.com/d3b-center/d3b-analysis-toolkit/pull/34
https://github.com/d3b-center/d3b-analysis-toolkit/pull/49

This is needed for the monthly histologies file release from @HuangXiaoyan0106. The model is such that Ops will run the bix-specific fields + toolkit annotations (subtyping, broad/short histology, cns_region) on a monthly basis or when we receive new patient data. 
cc @yuankunzhu, @zhangb1, @komalsrathi, @sickler-alex 

#### [Required] Project Infrastructure. Provide a link to the existing git repo / Cavatica project / HPC path. If a brand new project, provide title for the git repo / Cavatica project.

links above

**Comment by jharenza** [comment]
This is also in the base histologies PR we are working on which will be automated by xiaoyan - we should discuss

**Comment by aadamk** [comment]
Can you provide the link to the PR?

**Comment by jharenza** [comment]
Yes, [here](https://github.com/d3b-center/histologies-qc/pull/1) it is. It is being used currently to get v12 out the door, but I think Xiaoyan can likely just clone this workflow and adapt for the monthly and per-patient release. I have had to add probably 50 updates to the broad/short histology mapping here, so I really would rather keep the mapping in GitHub (whether it is a PR we make to a source of truth warehouse table that every workflow uses, or we all use this repo as source of truth - latter probably too clunky and time consuming). We can probably work with Nick to create these tables and we pull the files newly for these PRs, as we do for base adapt.

Note (unrelated to mapping), I recall asking if the monthly release does any filtering and was told no - so there might need to be a process for making sure we are not including data which is not visible (ie may have consent or quality issues). Here, we are using lists to add the new files - also not something I want to use long term but what we have now.

**Comment by aadamk** [comment]
To follow up here and from slack conversation, I agree with Nick maintaining the mapping tables. Then, Matt's code here can pull those tables and perform the mapping for the monthly release or real-time case. 

Looks like this mapping table [PR](https://github.com/d3b-center/d3b-dbt-study-transforms/pull/27/files#diff-225e861dcdb87c83eca4c6121de9012ffe097292b4fdeab74bf6022f9788beed) was closed a while back - was that due to the current CNS region ambiguity? Or were you waiting on updates to short histology? For the purposes of real-time cases, I'm fine with leveraging the current mapping knowledge (to assemble cohorts) with the understanding that there is a pending update.

**Comment by jharenza** [comment]
Ah, so this in particular was closed because Nick suggested Xiaoyan add them - she had added these to the warehouse, but I can't find where she did that. However, her manually adding them doesn't give us flexibility to make a PR to change, so I do think we should store in nick's table views. I also do not think the CNS region file here is very usable as those were very specific to v11 and now, they've basically been really mashed up again with updates to RedCap. I would suggest using the files from [here](https://github.com/d3b-center/histologies-qc/pull/1). 

P.S. I thought that the ops automated histologies file was a base file, but it appears it is just using the OPC table - I commented on a ticket to ask Xiaoyan to output the base file she is creating every week - the idea is to get this PR merged and do this creation weekly, which then would remove the need to create a separate workflow but let's see what it looks like. We made a lot of updates in this PR.

**Comment by HuangXiaoyan0106** [comment]
I'll follow the PR and figure out how to set it in the monthly release, and keep you posted.

**Comment by HuangXiaoyan0106** [comment]
> Ah, so this in particular was closed because Nick suggested Xiaoyan add them - she had added these to the warehouse, but I can't find where she did that.

@jharenza Sorry, I didn't add that information. But I will look into the PR that you updated.

**Comment by HuangXiaoyan0106** [comment]
> I think we need to have them somewhere in the WH - either by you or Nick - which is connected to github and the ability to PR an update to that table and have it auto push to update the WH table, similar to what nick does here: https://github.com/d3b-center/d3b-dbt-study-transforms - is that something you can set up or should we use nick's setup?

I'm not familiar with this and might need Nick's help.

**Comment by aadamk** [comment]
> I'm not familiar with this and might need Nick's help.

Given that this would require additional development work and a learning curve for @HuangXiaoyan0106, I advocate for continuing to use d3b-dbt-study-transforms via adapt as the central source of truth. 

Then, Matt can add back his code here in the toolkit, pulling that mapping table dynamically (in lieu of the static json file in the current implementation). Previously, Matt's short/broad mapping was embedded within the subtyping workflow, so adding this wf to a bix-ops scheduler that completes annotation for multiple fields (short/broad histology, mol subtype, cns_region when ready) would be a solution for the mtb side. There are currently histologies (base-adapt fields + germline sex, tumor purity, tumor ploidy) + data assembly files (ngs data) generated monthly for mtb, so the clinical reporting needs this annotation step synced for there to be a usable data product. 

> Ah, so this in particular was closed because Nick suggested Xiaoyan add them - she had added these to the warehouse, but I can't find where she did that. However, her manually adding them doesn't give us flexibility to make a PR to change, so I do think we should store in nick's table views.

So as a next step, can we re-open the PR with updated tables? This would unblock Matt to bring in his mapping script, which would query the warehouse table.

**Comment by jharenza** [comment]
>So as a next step, can we re-open the short/broad/cns region PR in study transforms with updated tables? This would unblock Matt to bring in his mapping script, which would query the warehouse table.

Yes - could you or matt. do it using the pathology mapping files here: https://github.com/d3b-center/histologies-qc/pull/1/files

I do not think CNS region will work in this way, or you can also feel free to regenerate this csv file using the latest base, but since the csv file was created post base - it was a unique cns_region to primary site map, it will constantly change based on the primary site constant changes - does that make sense? (vs a true mapping JSON file)
